tests/zenko_tests/node_tests/iam_policies/cloudserver/IAMUser.js

```diff
@@ -0,0 +1,257 @@
+const denyMdSearchPolicyName = "denyMDSearchPolicy";
+
+function createPolicy(action, isAllow = true, resource = "*") {
+    return JSON.stringify({
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Sid": "Stmt1644586763301",
+                "Action": [
+                    action,
+                ],
+                "Effect": (isAllow ? "Allow" : "Deny"),
+                "Resource": resource
+            }
+        ]
+    });
+}
+
+function metadataSearchResponseCode(userCredentials, bucketName, cb) {
+    return makeGETRequest(`/${bucketName}/?search=${encodeURIComponent('key LIKE "file"')}`, (err, res) => {
+        if (err)
+            return cb(err);
+        return cb(null, res.statusCode);
+    }, userCredentials)
+}
+
+describe('IAM Policies MetadataSearch', () => {
+
+    let userCredentials = {
+        accessKeyId: "",
+        secretAccessKey: "",
+    };
+
+    before(done => {
+        async.series([
+            next => s3.createBucket({Bucket: bucketName}, next),
+            next => s3.createBucket({Bucket: bucketName2}, next),
+            next => iam.createUser({UserName: userName}, next),
+            next => iam.createAccessKey({UserName: userName}, next)
+        ], (err, res) => {
+            assert.ifError(err);
+            userCredentials.accessKeyId = res[3].AccessKey.AccessKeyId;
+            userCredentials.secretAccessKey = res[3].AccessKey.SecretAccessKey;
+            done();
+        });
+    });
+
+    after(done => {
+        async.series([
+            next => iam.deleteUser({UserName: userName}, next),
+            next => s3.deleteBucket({Bucket: bucketName}, next),
+            next => s3.deleteBucket({Bucket: bucketName2}, next),
+        ], err => {
+            assert.ifError(err);
+            done();
+        });
+    })
+
+    it('should not be able to metadata search without the policy s3:MetadataSearch', done => {
+        return metadataSearchResponseCode(userCredentials, bucketName, (err, res) => {
+            if (err) {
+                assert.ifError(err)
+                return done(err)
+            }
+            assert.strictEqual(res, 403);
+            return done();
+        });
+    });
+
+    it('should be able to metadata search with the policy s3:MetadataSearch allowed', done => {
+        async.series([
+            next => iam.createPolicy({PolicyName: allowMdSearchPolicyName, PolicyDocument: createPolicy("s3:MetadataSearch")}, (err, res) => {
+                if (err)
+                    return next(err);
+                return iam.attachUserPolicy({UserName: userName, PolicyArn: res.Policy.Arn}, next);
+            }),
+            next => metadataSearchResponseCode(userCredentials, bucketName, (err, res) => {
+                    if (err) {
+                        assert.ifError(err)
+                        return next(err)
+                    }
+                    assert.strictEqual(res, 200);
```

Comment:
Can we also check here if we get the right information from the bucket? To make sure having an allowed permission to perform MD search is not affecting the behaviour of the API.

Reply:
Actually, the ft_tests are already done in cloudserver, I don't think we need to double check this, here we just want to check if the policy give the permissions to bucket search